datafusion/functions-aggregate/src/first_last.rs

```diff
@@ -652,7 +652,7 @@ impl Accumulator for LastValueAccumulator {
             // version in the new data:
             if !self.is_set
                 || self.requirement_satisfied
-                || compare_rows(&self.orderings, last_ordering, &sort_options)?.is_lt()
+                || compare_rows(&self.orderings, last_ordering, &sort_options)?.is_le()
```

Comment:
Just a note: when the aggregation is parallelized, then "true" last value may not be calculated anyway (if there is such a thing as a last value when scanning multiple files).